skycoin/skywire
# Improvements for the UI/UX of the manager

Did you run `make format && make check`?
The go code was not changed. `npm run lint` and `npm run build` were used.

Fixes #429

 Changes:	
- The procedure for updating the node list and node data was changed, so that now the data is managed by a centraliced service. This avoids the app always having to update the data after each navigation, which makes navigating some parts of the UI much faster.
- Now the code does not check the state of the login before each navigation, but just in relevant parts, which makes the navigation much faster in all parts of the app.
- Now the node list (with and without dmsg info) has pagination.
- An option for removing all offline visors was added to the visor list.
- Tab changes are now saved in the navigation history.
- The delay for retrying to get the data after an error was centralized in a configuration variable.
- Some other small improvements and fixes.

How to test this PR:
The first 2 points can be checked by navigating between the routing and apps tabs in the visor details page. With the old version it had to reload the data in each tab, but now the data should continue being available just after navigating.

For checking the pagination in the node list, the hypervisor must return a list of more than 40 connected visors.

The option for removing all offline visors can be found by pressing the new button at the top-right part of the visor list.

For testing the tab changes, simply navigate to a new tab and use the browser's back button for returning to the previous tab.
